ci/scripts/r_revdepcheck.sh

```diff
@@ -42,16 +75,26 @@ apt install -y libxml2-dev \
   libgeos-dev \
   libproj-dev
 
-pushd ${source_dir}
+
+# We have to be in source_dir so that cpp source detection works
+pushd $source_dir
 
 printenv
 
+# copy over cpp source
+make sync-cpp
+
 # By default, aws-sdk tries to contact a non-existing local ip host
 # to retrieve metadata. Disable this so that S3FileSystem tests run faster.
 export AWS_EC2_METADATA_DISABLED=TRUE
 
 # Set crancache dir so we can cache it
-export CRANCACHE_DIR="/arrow/.crancache"
+export CRANCACHE_DIR="${1}/.crancache"
+
+# Don't use system boost to prevent issues with missing components 
+export EXTRA_CMAKE_FLAGS='-DBoost_SOURCE=BUNDLED'
```

Comment:
This might be unfounded, but I have a small fear that adding extra flags here means we're drifting away from how this is run on CRAN. This _probably_ doesn't matter. But is it possible to run this build without the extra flags?

Reply:
This just fixes issues with half installed boost components, I think it might be fixed by now, let me check

Reply:
After testing around for a while I now remember why this is necessary, one of revdeps installs another version of boost (missing components we need) that is found before the fulll system install and thus causes the build to fail with a missing boost include dir. I will add a comment better explaining the situation but this is unique to this build and has no impact on the revdepcheck itself :+1: